### PR TITLE
FIX: import from correct mpl backend by qt version

### PR DESCRIPTION
**Pull request by tacaswell**
This enable enaml + matplotlib to work with pyqt5.

### CODE REVIEW

File: enaml/qt/qt_mpl_canvas.py [hunk 1]
```diff
@@ -9,18 +9,26 @@
...
-from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg
-try:
-    from matplotlib.backends.backend_qt4agg import (NavigationToolbar2QTAgg
-                                                    as NavigationToolbar2QT)
-except ImportError:
-    from matplotlib.backends.backend_qt4agg import NavigationToolbar2QT
-
-from .QtCore import Qt
+from qtpy import QT_VERSION
+from .QtCore import Qt, PYQT_VERSION_STR
```
Comment (MatthieuDartiailh) [inline enaml/qt/qt_mpl_canvas.py]: PYQT_VERSION_STR is unused so please remove it. Also you can avoid importing QT_VERSION from qtpy by importing \_\_version\_\_ or \_\_version_info\_\_(tuple of int) from .QtCore which I would prefer.

File: enaml/qt/qt_mpl_canvas.py [hunk 2]
```diff
@@ -9,18 +9,26 @@
...
+    try:
+        from matplotlib.backends.backend_qt4agg import (NavigationToolbar2QTAgg
+                                                        as NavigationToolbar2QT)
+    except ImportError:
+        from matplotlib.backends.backend_qt4agg import NavigationToolbar2QT
+elif QT_VERSION[0] == '5':
+    from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+    from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT
+else:
+    raise RuntimeError('Using qtpy.QT_VERSION == {} which we do not know how '
```
Comment (MatthieuDartiailh) [inline enaml/qt/qt_mpl_canvas.py]: Here I find the message a bit confusing. I would say: 
```python
"No known Matplotlib backend for qt version {__version__} (as reported by qtpy.QT_VERSION)"
```
I am not convinced mentioning qtpy here really helps but I am not opposed to keeping the mention

**Review by MatthieuDartiailh — CHANGES_REQUESTED** [review]
Some minor changes otherwise LGTM

File: enaml/qt/qt_mpl_canvas.py (hunk 1 above)
Comment (tacaswell) [reply enaml/qt/qt_mpl_canvas.py]: Are those the version of qt or of the wrapper?

Comment (MatthieuDartiailh) [reply enaml/qt/qt_mpl_canvas.py]: Actually \_\_version\_\_ is just an alias of QT_VERSION exported by .QtCore

**Comment by MatthieuDartiailh** [comment]
@tacaswell do you have time to address the minor comments ?

**Comment by MatthieuDartiailh** [comment]
ping @tacaswell

**Comment by tacaswell** [comment]
Sorry, this got buried at the bottom of my 2.4k notifications :disappointed: 

Fixed and force-pushed.

**Review by MatthieuDartiailh — APPROVED** [review]

**Comment by MatthieuDartiailh** [comment]
Thanks @tacaswell merging now. I will try to get a fix for qt console and I will release a new version afterwards.

**Comment by MatthieuDartiailh** [comment]
@tacaswell I get that you are awfully busy but you may be interested in doing a quick review of the PR opened on this repo as they may be of interest to you, and I would really appreciate it.

**Comment by tacaswell** [comment]
Any in particular or just in general?

**Comment by MatthieuDartiailh** [comment]
I though about the one about the qtconsole that I merged two days ago, in case you are familiar with it ? Otherwise I would really like some feedback on the setup commands. I need to improve the documentation about how to use it in a way allowing setuptools to install enaml before it is needed by the package, but otherwise I think it would be a nice improvement.
I think I will make a new release with the matplotlib fix, the ipython fix and the setup improvement once it is merged.

**Pull request merged**